docs/content/docs/dev/table/data_stream_api.md

```diff
@@ -0,0 +1,2106 @@
+
+Note that only non-updating tables are supported by `toDataStream`. Usually, time-based operations
+such as windows, interval joins, or the `MATCH_RECOGNIZE` clause are a good fit for insert-only
+pipelines next to simple operations like projections and filters. Pipelines with operations that
+produce updates can use `toChangelogStream`.
+
+{{< top >}}
+
+Handling of Changelog Streams
+-----------------------------
+
+Internally, Flink's table runtime is a changelog processor. The concepts page describes how
+[dynamic tables and streams are related]({{< ref "docs/dev/table/concepts/dynamic_tables" >}})
+with each other.
+
+A `StreamTableEnvironment` offers the following methods to expose these _change data capture_ (CDC) functionalities:
+
+- `fromChangelogStream(DataStream)`: Interprets a stream of changelog entries as a table. The stream
+record type must be `org.apache.flink.types.Row` as the `RowKind` flag that is contained in
+every record is evaluated during runtime. Event-time and watermarks are not propagated by default.
+This method expects a changelog containing all kinds of changes (enumerated in `RowKind`) as the
+default `ChangelogMode`.
+
+- `fromChangelogStream(DataStream, Schema)`: Allows to define a schema for the `DataStream` similar
+to `fromDataStream(DataStream, Schema)`. Otherwise the semantics are equal to `fromChangelogStream(DataStream)`.
+
+- `fromChangelogStream(DataStream, Schema, ChangelogMode)`: Gives full control about how to interpret a
+stream as a changelog. The passed `ChangelogMode` helps the planner to distinguish between _insert-only_,
+_upsert_, or _retract_ behavior.
+
+- `toChangelogStream(Table)`: Reverse operation of `fromChangelogStream(DataStream)`. It produces a
+stream with instances of `org.apache.flink.types.Row` and sets the `RowKind` flag for every record
+during runtime. All kinds of updating tables are supported by this method. If the input
+table contains a single rowtime column, it will be propagated into a stream record's timestamp. Watermarks
+will be propagated as well.
+
+- `toChangelogStream(Table, Schema)`: Reverse operation of `fromChangelogStream(DataStream, Schema)`.
+The method can enrich the produced column data types. The planner might insert implicit casts if necessary.
+It is possible to write out the rowtime as a metadata column.
+
+- `toChangelogStream(Table, Schema, ChangelogMode)`: Gives full control about how to convert a table
+to a changelog stream. The passed `ChangelogMode` helps the planner to distinguish between _insert-only_,
+_upsert_, or _retract_ behavior.
+
+From a Table API's perspective, converting from and to DataStream API is similar to reading from or
+writing to a virtual table connector that has been defined using a [`CREATE TABLE` DDL]({{< ref "docs/dev/table/sql/create" >}}#create-table)
+in SQL.
+
+Because `fromChangelogStream` behaves similar to `fromDataStream`, we recommend to read
+the [previous section](#handling-of-insert-only-streams) before.
```

Comment:
```suggestion
Because `fromChangelogStream` behaves similar to `fromDataStream`, we recommend reading
the [previous section](#handling-of-insert-only-streams) before continuing here.
```